build: Add `reset` script

tomi opened the pull request:
## Summary

We often face situations where a build will fail because some cache somewhere is messing up with the build. This PR adds a `nuke` script that can be used to delete all files that might be out of date (i.e. node_modules, dist files, .turbo cache, tsbuildinfo files, etc). It is a more destructive version from `clean`, which doesn't touch e.g. node_modules.


## Review / Merge checklist

- [ ] PR title and summary are descriptive. ([conventions](../blob/master/.github/pull_request_title_conventions.md)) <!--
   **Remember, the title automatically goes into the changelog.
   Use `(no-changelog)` otherwise.**
-->
- [ ] [Docs updated](https://github.com/n8n-io/n8n-docs) or follow-up ticket created.
- [ ] Tests included. <!--
   A bug is not considered fixed, unless a test is added to prevent it from happening again.
   A feature is not complete without tests.
-->
- [ ] PR Labeled with `release/backport` (if the PR is an urgent fix that needs to be backported)

tomi committed:
build: Add nuke script

We often face situations where a build will fail because some cache somewhere
is messing up with the build. This PR adds a `nuke` script that can be used to
delete all files that might be out of date (i.e. node_modules, dist files, .turbo
cache, tsbuildinfo files, etc). It is a more destructive version from `clean`, which
doesn't touch e.g. node_modules.

tomi committed:
Also nuke stuff in the root directory

netroy reviewed (commented):
instead of maintaining a list of purgable files/folder in every package, could we use `git clean -xd` instead (with `--exclude `to skip deleting certain files/folders)?

tomi committed:
Revert "Also nuke stuff in the root directory"

This reverts commit 67b28bba491ed15a50f32bf4bda81eeb32a91269.

tomi committed:
Revert "build: Add nuke script"

This reverts commit 8661a8f2e20de91cf75b26aefafbcfa8271dcc83.

tomi committed:
build: Add reset script

tomi commented:
> instead of maintaining a list of purgable files/folder in every package, could we use `git clean -xd` instead (with `--exclude `to skip deleting certain files/folders)?

@netroy I updated the script to be a bit more sophisticated. Added`.vscode/settings.json` and `.env` files to exclude. Anything else?

tomi committed:
Update scripts/reset.mjs

Co-authored-by: कारतोफ्फेलस्क्रिप्ट™ <netroy@users.noreply.github.com>

tomi committed:
Exclude .idea from deletion

netroy commented on scripts/reset.mjs:
```diff
@@ -0,0 +1,29 @@
+// Resets the repository by deleting all untracked files except for few exceptions.
```
maybe we should add `process.env.FORCE_COLOR='1'` to pipe [subprocess TTY colors correctly](https://google.github.io/zx/known-issues#colors-in-subprocess)

netroy reviewed (commented):
this should work, but has one minor issue that if someone runs `pnpm reset` before running `pnpm i`, this script will error out.

tomi committed:
Fix colors

tomi committed:
Ensure zx is installed before executing reset

tomi commented:
> this should work, but has one minor issue that if someone runs `pnpm reset` before running `pnpm i`, this script will error out.

Fixed this by ensuring zx is available. Not necessary ideal to run `pnpm install` to just delete the node_modules and run `pnpm install` again, but it works

janober commented:
Got released with [`n8n@1.58.0`](https://github.com/n8n-io/n8n/releases/tag/n8n%401.58.0)
